core/shunting.ml

```diff
@@ -0,0 +1,296 @@
+(* An implementation of Dijkstra's Shunting Yard algorithm. *)
+open Sugartypes
+open SourceCode.WithPos
+open Operators
+
+let name_of_binop = function
```

Comment:
Does this really belong here? I would have thought that this would have belonged more in either `Operators` or `Sugartypes` rather than a transformation pass.

Reply:
I'd like to keep it local for now. It will go away with the introduction of hygienic names.

Reply:
OK, if this is a conscious choice then that's fine.